stardis/base.py

```diff
@@ -110,64 +102,55 @@ class STARDISOutput:
-    doppler_widths : numpy.ndarray
-        Array of shape (no_of_lines, no_of_depth_points). Doppler width of each
-        line at each depth point.
-    F_nu : astropy.units.Quantity
-        Array of shape (no_of_depth points, no_of_frequencies). Output flux with
-        respect to frequency at each depth point for each frequency. Units of erg/s/cm^2/Hz.
-    F_lambda : astropy.units.Quantity
-        Array of shape (no_of_depth_points, no_of_frequencies). Output flux with
-        respect to wavelength at each depth point for each wavelength. Units of erg/s/cm^2/Angstrom.
+    stellar_plasma : tardis.plasma.base.BasePlasma
+    stellar_radiation_field : stardis.radiation_field.radiation_field.RadiationField
+        contains the following attributes:
+            frequencies : astropy.units.Quantity
+                Frequencies of the radiation field.
+            source_function : stardis.radiation_field.source_function
+            opacities : stardis.radiation_field.opacities
+                Stardis opacities object. Contains the opacities contributed by the stellar atmosphere.
+            F_nu : numpy.ndarray
+                Radiation field fluxes at each frequency at each depth point.
+    nus : astropy.units.Quantity
+        Numpy array of frequencies used for spectrum with units of Hz.
+    lambdas : astropy.units.Quantity
+        Numpy array of wavelengths used for spectrum with units of Angstroms.
     spectrum_nu : astropy.units.Quantity
         Output flux with respect to frequency at the outermost depth point for each
         frequency. Units of erg/s/cm^2/Hz.
     spectrum_lambda : astropy.units.Quantity
         Output flux with respect to wavelength at the outermost depth point for each
         wavelength. Units of erg/s/cm^2/Angstrom.
-    nus : astropy.units.Quantity
-        Numpy array of frequencies used for spectrum with units of Hz.
-    lambdas : astropy.units.Quantity
-        Numpy array of wavelengths used for spectrum with units of Angstroms.
     """
 
-    ###TODO: Instead of returning all these various quantities of the radiation, simply return
-    # the radiation field with class properties that return useful quantities such as spectrum lambda and lambdas.
-
     def __init__(
-        self, stellar_plasma, stellar_model, alphas, gammas, doppler_widths, F_nu, nus
+        self,
+        result_options,
+        stellar_model,
+        stellar_plasma,
+        stellar_radiation_field,
     ):
-        self.stellar_plasma = stellar_plasma
-        self.stellar_model = stellar_model
-        self.alphas = alphas
-        self.gammas = gammas
-        self.doppler_widths = doppler_widths
-
-        self.nus = nus
-        self.lambdas = nus.to(u.AA, u.spectral())
-
-        self.F_nu = F_nu * u.erg / u.s / u.cm**2 / u.Hz
-        self.F_lambda = (self.F_nu * nus / self.lambdas).to(
-            u.erg / u.s / u.cm**2 / u.AA
-        )
+        if result_options.return_model == True:
```

Comment:
The explicit `== True` might be slightly overkill here

Reply:
Yup, reasonable. I've removed it.